core/main.tf

```
@@ -18,3 +18,29 @@ module "cloudwatch" {
 
   namespace = local.namespace
 }
+
+module "rds" {
+  source = "../modules/rds"
+
+  namespace = local.namespace
+
+  vpc_security_group_ids = module.sercurity_group.rds_security_groups_ids
+  vpc_id                 = module.vpc.vpc_id
+
+  subnets = module.vpc.private_subnets
+
+  instance_class = local.rds_instance_class
+  database_name  = var.rds_database_name
+  username       = var.rds_username
+  password       = var.rds_password
+
+  autoscaling_min_capacity = var.rds_autoscaling_min_capacity
+  autoscaling_max_capacity = var.rds_autoscaling_max_capacity
+}
+
+module "sercurity_group" {
+  source = "../modules/security_group"
+
+  namespace = local.namespace
+  vpc_id    = module.vpc.vpc_id
+}
```

```suggestion
module "security_group" {
  source = "../modules/security_group"

  namespace = local.namespace
  vpc_id    = module.vpc.vpc_id
}
```

Do we need to create the `security_group` module yet, and thus use it here? Or can we use it when we add ALB

The argument `vpc_security_group_ids` on RDS is `optional` and the RDS security group you created doesn't have any rules - either from resource `aws_security_group_rule` or defined here with key `ingress`

---

from my understanding, the security group assigned to the RDS instance will allow traffic from other services within the VPC (can access the RDS instance).
We don't specify any inbound/outbound traffic since we don't want any access to our RDS instances.